component/src/main/java/org/wso2/extension/siddhi/store/influxdb/InfluxDBStore.java

```diff
@@ -0,0 +1,565 @@
+                } else {
+                    condition = condition.replaceFirst(Pattern.quote("'?'"), myObject.toString());
+                }
+            }
+            for (Map.Entry<Integer, Object> entry : constantMap.entrySet()) {
+                Object myObject = entry.getValue();
+
+                if (myObject instanceof String) {
+                    condition = condition.replaceFirst(Pattern.quote("*"), myObject.toString());
+                } else {
+                    condition = condition.replaceFirst(Pattern.quote("'*'"), myObject.toString());
+                }
+            }
+
+            findCondition = line.toString();
+            findCondition = findCondition + condition;
+        }
+
+        influxdb.setDatabase(database);
+        query = new Query(findCondition, database);
+        influxdb.query(query);
+        queryResult = influxdb.query(query);
+        return queryResult;
+    }
+
+    public void writePoints(List<Object[]> list) {
+
+        for (Object[] record : list) {
+            Map<String, String> insertTagMap = InfluxDBTableUtils.mapTagValuesToAttributes(record,
+                    this.attributeNames, this.tagPositions);
+            Map<String, Object> insertFieldMap = InfluxDBTableUtils.mapFieldValuesToAttributes(record,
+                    this.attributeNames, this.tagPositions);
+            long times = InfluxDBTableUtils.mapTimeToAttributeValue(record, this.tableName,
+                    this.timePosition);
+
+            influxdb.setDatabase(database);
+            BatchPoints batchPoints = BatchPoints.database(database).build();
+            Point point = Point.measurement(tableName)
+                    .time(times, TimeUnit.MILLISECONDS)
+                    .tag(insertTagMap)
+                    .fields(insertFieldMap)
+                    .build();
+            batchPoints.point(point);
+            influxdb.write(batchPoints);
+        }
+    }
+
+    public boolean checkDatabaseExists(String dbName) {
+
+        boolean isExists = false;
+        Query query = new Query("SHOW DATABASES", database);
+        QueryResult queryResult = influxdb.query(query);
+        List<List<Object>> databaseList = queryResult.getResults()
+                .get(0)
+                .getSeries()
+                .get(0)
+                .getValues();
+
+        for (int i = 0; i < databaseList.size(); i++) {
+            if (databaseList.get(i).get(0).equals(dbName)) {
+                isExists = true;
+            } else {
+                isExists = false;
+            }
+        }
+
+        return isExists;
+    }
+
+    @Override
+    protected RecordIterator<Object[]> query(Map<String, Object> parameterMap, CompiledCondition compiledCondition,
+                                             CompiledSelection compiledSelection, Attribute[] outputAttributes)
+            throws ConnectionUnavailableException {
+
+        return null;
```

Comment:
We cant query influx db as well?

Reply:
https://github.com/wso2-extensions/siddhi-store-influxdb/pull/1/commits/9234a75137029d90089406fe0c38319a27eedc0d